### PR TITLE
blockedit: correctly handle cursor when appending text

### LINKED ISSUE #8288: Appending text in visual block mode doesn't work properly when cursor is moved
**Describe the bug**
When appending text in visual block mode, if the cursor is moved before the
start of the visual block, then incorrect text is inserted.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Run `vim --clean`
2. Add the following text to an empty buffer.
        aaa
        bbb
        ccc
3. Execute the following commands:
       `gg$<C-V>2jA<Left>x<Esc>`

**Expected behavior**
The following lines are expected after the edit:
```
       aaxa
       bbxb
       ccxc
```

But the following lines are present after the edit:
```
       aaxa
       bbba
       ccca
```

The second and third lines are not correct.

**Environment (please complete the following information):**
 - Vim version 8.2.2911
 - OS: Ubuntu 21.04
 - Terminal: Gnome-terminal

Additional details:
A test for this is already added to the Test_visual_block_mode() function in test_visual.vim.

### PR DESCRIPTION
**Pull request by chrisbra**
closes #8288

### CODE REVIEW

File: src/testdir/test_blockedit.vim [hunk 1]
```diff
@@ -28,4 +28,12 @@ func Test_blockinsert_delete()
   bwipe!
 endfunc
 
+func Test_blockappend_eol_cursor()
```
Comment (yegappan) [inline src/testdir/test_blockedit.vim:31]: There is a commented out test for this in the Test_visual_block_mode() function
in test_visual.vim (search for BUG).

Comment (chrisbra) [reply src/testdir/test_blockedit.vim:31]: ah, missed that part of the message.

Comment (chrisbra) [reply src/testdir/test_blockedit.vim:31]: BTW: I remember always having wondered, why there is the condition for `bd.is_MAX`. I hope this does not cause any problem.s

**Review by yegappan — APPROVED** [review]

**Comment by brammool** [comment]
So, is the "needs more work" taken care of now?

**Comment by chrisbra** [comment]
it works for the tests I threw at it. Problem is, if you move with the cursor out of the visual selection.  For now I abort the block edit.

**Comment by brammool** [comment]
Mostly when moving the cursor while doing a block insert or append, we can abort the block edit.  Since it's hard to tell what exactly was inserted and how to repeat that.  Since you fixed this specific case I'll include it, but the code is getting more and more complex if we try to handle other cases.  So perhaps just do this one and don't worry about other cases.

File: src/testdir/test_visual.vim [hunk 2]
```diff
@@ -806,11 +806,8 @@ func Test_visual_block_mode()
   %d _
   call setline(1, ['aaa', 'bbb', 'ccc'])
   exe "normal $\<C-V>2jA\<Left>x"
-  " BUG: Instead of adding x as the third character in all the three lines,
-  " 'a' is added in the second and third lines at the end. This bug is not
-  " reproducible if this operation is performed manually.
-  "call assert_equal(['aaxa', 'bbxb', 'ccxc'], getline(1, '$'))
-  call assert_equal(['aaxa', 'bbba', 'ccca'], getline(1, '$'))
+  call assert_equal(['aaxa', 'bbxb', 'ccxc'], getline(1, '$'))
+  "call assert_equal(['aaxa', 'bbba', 'ccca'], getline(1, '$'))
```
Comment (dpelle) [inline src/testdir/test_visual.vim:810]: You uncommentted out the first `assert_equal` (which seems good) but you commented out the 2nd one.
Should the 2nd assert really be commented out?  If yes, we probably need  a comment to say why.  It's not clear to me at least but I did not spend much time looking at the change.

Comment (chrisbra) [reply src/testdir/test_visual.vim:810]: Yes, that commented second one is of course wrong and should be removed

Comment (chrisbra) [reply src/testdir/test_visual.vim:810]: ah, Bram had already fixed it.

**Pull request closed without merging**